src/app/campaign/flight/flight.container.ts

```diff
@@ -85,4 +98,24 @@ export class FlightContainerComponent implements OnInit, OnDestroy {
     this.campaignStoreService.loadAvailability(flight);
     this.currentInventoryUri$.next(flight.set_inventory_uri);
   }
+
+  flightDuplicate(flight: Flight) {
+    const localFlight: Flight = { ...flight, name: `${flight.name} (Copy)` };
+    this.campaignStoreService.campaignFirst$.subscribe(state => {
+      const flightId = Date.now();
+      this.campaignStoreService.setFlight({ localFlight, changed: true, valid: true }, flightId);
+
+      const campaignId = state.remoteCampaign ? state.remoteCampaign.id : 'new';
+      this.router.navigate(['/campaign', campaignId, 'flight', flightId]);
+    });
+  }
+
+  flightDeleteToggle() {
+    this.campaignStoreService.campaignFirst$.subscribe(state => {
+      const currentState = state.flights[this.currentFlightId];
+      const newState = { ...currentState, changed: true, softDeleted: !!!currentState.softDeleted };
```

Comment:
triple `!!!`? is this making sure it's extra soft deleted?

Reply:
Ha! As an optional property, it's sometimes `undefined`, so for some reason I thought it needed to be coerced this way. It doesn't.